HW05.Rmd

```diff
@@ -0,0 +1,90 @@
+---
+title: "HW05"
+author: "Xiaoshan Ke"
+date: "8/1/2020"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(ggplot2)
+library(dplyr)
+library(tidyr)
+library(stringr)
+```
+
+### HW Assignment
+
+This week the goal is to practice tidying data and joining files together. Similar to last week you can practice these skills with data you already have or you can follow with the two (probably pretty short) HW assignments below.
+
+## 1. Tidying the dad_mom file
+
+In this repo, you'll find a dad_mom.txt file that isn't particularly tidy. Clean this up so there are only 4 columns 
+Tidy this data frame so that it adheres to the tidy data principles:
+
+    Each variable must have its own column.
+    Each observation must have its own row.
+    Each value must have its own cell.
+    
+```{r}
+dad_mom <- read.delim("dad_mom.txt")
+clean_up <- dad_mom %>%
+  gather(key = "key", value = "value", name_dad:income_mom) %>%
+  separate(key, into = c("var","dad_or_mom"), convert = TRUE) %>%
+  spread(var, value) 
```

Comment:
This is an excellent example of using functions gather( ), separate( ), and spread( ) together to reshape data set!

Reply:
I think this is really a good way to do it! I added to mine for documentation.